### ISSUE TITLE
Cannot find module node_modules/babel-loader/lib/index.js?cacheDirectory

### ISSUE THREAD

**Issue by shaodahong** [babel/babel-loader#691]
**I'm submitting a bug report**

**Webpack Version:**
"webpack": "^4.20.2"

**Babel Core Version**:
"@babel/core": "^7.0.0"

**Babel Loader Version**:
"babel-loader": "^8.0.0"

**Please tell us about your environment:**
OSX 10.14

webpack config:

```js
new HappyPack({
  id: 'jsx',
  threads: 4,
  loaders: [
    {
      loader: 'babel-loader?cacheDirectory',
      options: require('./babel.config'),
      exclude: ['/node_modules/']
    }
  ]
}),
```

Error Image:

![image](https://user-images.githubusercontent.com/14831261/46131623-b98cbe80-c26e-11e8-8ecd-19e3ff402cd1.png)

Node_modules folder have `babel-loader`, when I remove cacheDirectory option, it's ok,

**Comment by nicolo-ribaudo** [comment]
Looking at the happypack document, I think that you should use

```
loader: 'babel-loader',
query:  '?cacheDirectory',
```

https://github.com/amireh/happypack/blob/master/README.md#loaders-array

**Comment by shaodahong** [comment]
@nicolo-ribaudo just error

```js
new HappyPack({
  id: 'jsx',
  threads: 4,
  loaders: [
    {
      loader: 'babel-loader',
      query:  '?cacheDirectory',
      options: require('./babel.config'),
      exclude: ['/node_modules/']
    },
    {
      loader: path.join(__dirname, '/auto-prod-filter-loader')
    },
    {
      loader: 'eslint-loader',
      exclude: ['/node_modules/']
    }
  ]
}),
```

![image](https://user-images.githubusercontent.com/14831261/46134430-6e29de80-c275-11e8-95e7-9603d2534c43.png)

**Comment by shaodahong** [comment]
I try to set `cacheDirectory` in `babel.config.js`, it's ok

```js
module.exports = {
  cacheDirectory: true,
  presets: [
    '@babel/preset-react'
}
```

**Comment by loganfsmyth** [comment]
`babel.config.js` is a config file that Babel itself loads, you should not be `require`ing it yourself, and `cacheDirectory` should not be inside that file, it is only an option for `babel-loader`, not for Babel itself.

**Comment by shaodahong** [comment]
@loganfsmyth yes, but I can't set `cacheDirectory ` option, the console has an error when running application

**Comment by shaodahong** [comment]
in `babel-loader` index.js

```js
const {
  cacheDirectory = null
} = loaderOptions;
```

1. `cacheDirectory` from loader options
2. and after that await cache
3. `delete programmaticOptions.cacheDirectory`

so `babel.config.js` babel-loader will handle it， right?

**Comment by loganfsmyth** [comment]
Can you try just doing
```
    {
      loader: 'babel-loader',
      options: {
        cacheDirectory: true,
      }
      exclude: ['/node_modules/']
    },
```
Also FYI 
```
      exclude: ['/node_modules/']
```
most likely does nothing, since the string-based exclude is a path, not a regex.

**Comment by shaodahong** [comment]
@loganfsmyth I can't do, this project is a public package, so babel config is private, I don't want to expose, so If `babel-loader` options don't set babel config, they can be search in user current workspace, this is what I don't want to see
